php_phongo.c

```diff
@@ -342,7 +341,6 @@ bool phongo_query_init(php_phongo_query_t *query, zval *filter, zval *options TS
 				return false;
 			}
 
-			convert_to_array_ex(&projection);
 			query->selector = bson_new();
```

Comment:
The projection and query document (i.e. what we merge modifiers into above) didn't need this treatment since they are top-level documents and you encode them as such later on. Sort was the unique case where it gets stuffed in an `$orderby` field and could be encoded as an array inadvertently.

I do mean to split these changes up into separate commits, but the `convert_to_array_ex()` deletions here and for modifiers are basically just removing dead code.